### PR TITLE
[board] Add modm:board:nucleo-f072rb

### PR DESCRIPTION
**Pull request by TomSaw**
Had this board lying around. Quickly copy-pasted and tested the board files.

### CODE REVIEW

File: src/modm/board/nucleo_f072rb/board.hpp [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+
+inline void
+initialize()
+{
+	SystemClock::enable();
+	SysTickTimer::initialize<SystemClock>();
+
+	// stlink::Uart::connect<stlink::Tx::Tx, stlink::Rx::Rx>();
+	// stlink::Uart::initialize<SystemClock, 115200_Bd>();
```
Comment (rleh) [inline src/modm/board/nucleo_f072rb/board.hpp]: Should not be commented out.

File: src/modm/board/nucleo_f072rb/board.xml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    <repository>
+      <path>../../../../repo.lb</path>
+    </repository>
+  </repositories>
+
+  <options>
+    <option name="modm:target">stm32f072rbt6</option>
+
+    <option name="modm:platform:uart:1:buffer.tx">256</option>
+    <option name="modm:platform:cortex-m:main_stack_size">1024</option>
```
Comment (rleh) [inline src/modm/board/nucleo_f072rb/board.xml]: Is there a reason to set the main stack size in this BSP?

**Review by rleh — DISMISSED** [review]
As the CI also says, you need to update the README:
```bash
python3 tools/scripts/synchronize_docs.py
```

---

Also please add/copy (at least) an minimal *blink* example so that the BSP gets tested.
You also have to add the BSP to the [CI configuration](https://github.com/modm-io/modm/blob/develop/.github/workflows/linux.yml#L93).

**Comment by TomSaw** [comment]
Thx @rleh for the review. I've fixed this but must check later in hardware.

File: src/modm/board/nucleo_f072rb/board.hpp [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+		// Switch to the 48MHz clock
+		Rcc::enableSystemClock(Rcc::SystemClockSource::InternalClockMHz48);
+		// update frequencies for busy-wait delay functions
+		Rcc::updateCoreFrequency<Frequency>();
+
+		return true;
+	}
+};
+
+// Arduino Nano Footprint
```
Comment (salkinium) [inline src/modm/board/nucleo_f072rb/board.hpp]: Uno, no?

File: src/modm/board/nucleo_f072rb/board.hpp [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+	static constexpr uint32_t Timer15 = Apb;
+	static constexpr uint32_t Timer16 = Apb;
+	static constexpr uint32_t Timer17 = Apb;
+
+	static constexpr uint32_t Usb = 48_MHz;
+
+	static bool inline
+	enable()
+	{
+		// Enable the interal 48MHz clock
```
Comment (salkinium) [inline src/modm/board/nucleo_f072rb/board.hpp]:
```suggestion
		// Enable the internal 48MHz clock
```

**Review by salkinium — APPROVED** [review]

File: src/modm/board/nucleo_f072rb/board.hpp (hunk 3 above)
Comment (TomSaw) [reply src/modm/board/nucleo_f072rb/board.hpp]: Si, Uno

File: src/modm/board/nucleo_f031k6/board.hpp [hunk 5]
```diff
@@ -71,7 +71,7 @@ struct SystemClock {
 	}
 };
 
-// Arduino Nano Footprint
+// Arduino Uno Footprint
```
Comment (salkinium) [inline src/modm/board/nucleo_f031k6/board.hpp]: Nano, no? (Nucleo32 vs Nucleo64)

File: src/modm/board/nucleo_f042k6/board.hpp [hunk 6]
```diff
@@ -74,7 +74,7 @@ struct SystemClock {
 	}
 };
 
-// Arduino Nano Footprint
+// Arduino Uno Footprint
```
Comment (salkinium) [inline src/modm/board/nucleo_f042k6/board.hpp]: Same.

File: src/modm/board/nucleo_f303k8/board.hpp [hunk 7]
```diff
@@ -85,7 +85,7 @@ struct SystemClock {
 	}
 };
 
-// Arduino Nano Footprint
+// Arduino Uno Footprint
```
Comment (salkinium) [inline src/modm/board/nucleo_f303k8/board.hpp]: .

File: src/modm/board/nucleo_l031k6/board.hpp [hunk 8]
```diff
@@ -78,7 +78,7 @@ struct SystemClock
 	}
 };
 
-// Arduino Nano Footprint
+// Arduino Uno Footprint
```
Comment (salkinium) [inline src/modm/board/nucleo_l031k6/board.hpp]: .

File: src/modm/board/nucleo_l432kc/board.hpp [hunk 9]
```diff
@@ -67,7 +67,7 @@ struct SystemClock {
 	}
 };
 
-// Arduino Nano Footprint
+// Arduino Uno Footprint
```
Comment (salkinium) [inline src/modm/board/nucleo_l432kc/board.hpp]: all the *k identifiers point to Nucleo32

Comment (TomSaw) [reply src/modm/board/nucleo_l432kc/board.hpp]: I've just copied this... so already ctrl+h't all of these ;)

Comment (salkinium) [reply src/modm/board/nucleo_l432kc/board.hpp]: I… don't understand your response?

- [Nucleo32 has Arduino™ Nano V3 connectivity](https://www.st.com/content/st_com/en/products/evaluation-tools/product-evaluation-tools/mcu-mpu-eval-tools/stm32-mcu-mpu-eval-tools/stm32-nucleo-boards/nucleo-g031k8.html).
- [Nucleo64 has ARDUINO® Uno V3 connectivity](https://www.st.com/content/st_com/en/products/evaluation-tools/product-evaluation-tools/mcu-mpu-eval-tools/stm32-mcu-mpu-eval-tools/stm32-nucleo-boards/nucleo-f072rb.html).

**Comment by TomSaw** [comment]
> As the CI also says, you need to update the README:
> 
> ```shell
> python3 tools/scripts/synchronize_docs.py
> ```
> 
> Also please add/copy (at least) an minimal _blink_ example so that the BSP gets tested.
> You also have to add the BSP to the [CI configuration](https://github.com/modm-io/modm/blob/develop/.github/workflows/linux.yml#L93).

I think it's all done. Please approve changes @rleh

**Comment by rleh** [comment]
Sorry, I was too busy last week...

**Pull request merged**